src/components/scenes/LoginScene.tsx

```diff
@@ -33,136 +32,135 @@ import { LoadingScene } from './LoadingScene'
           .catch(showError)
       }
       if (YOLO_PASSWORD != null) {
         context
           .loginWithPassword(YOLO_USERNAME, YOLO_PASSWORD)
-          .then(account => initializeAccount(navigation, account, dummyTouchIdInfo))
+          .then(async account => {
+            await dispatch(initializeAccount(navigation, account, dummyTouchIdInfo))
+          })
           .catch(showError)
       }
     }
+  }, [context, dispatch, navigation])
+
+  React.useEffect(() => {
+    if (pendingDeepLink != null && pendingDeepLink.type === 'passwordRecovery') {
+      // Log out if necessary:
+      if (account.loggedIn) {
+        dispatch(logoutRequest(navigation))
+      }
+
+      // Pass the link to our component:
+      const { passwordRecoveryKey } = pendingDeepLink
+      setPasswordRecoveryKey(passwordRecoveryKey)
+      setCounter(counter => counter + 1)
+      dispatch({ type: 'DEEP_LINK_HANDLED' })
+    }
+  }, [account, dispatch, navigation, pendingDeepLink])
+
+  const checkForUpdates = useHandler(async () => {
     const response = await checkVersion()
-    const skipUpdate = (await this.getSkipUpdate()) === response.version
+    const skipUpdate = (await disklet.getText('ignoreUpdate.json').catch(() => '')) === response.version
     if (response.needsUpdate && !skipUpdate) {
       Airship.show(bridge => (
         <UpdateModal
           bridge={bridge}
           onSkip={() => {
-            this.props.disklet.setText('ignoreUpdate.json', response.version)
+            disklet.setText('ignoreUpdate.json', response.version)
             bridge.resolve()
           }}
         />
       ))
     }
-  }
-
-  componentDidUpdate(oldProps: Props) {
-    const { account, pendingDeepLink, theme } = this.props
-    const backgroundImageServerUrl = pickRandom(theme.backgroundImageServerUrls)
-
-    getBackgroundImage(this.props.disklet, backgroundImageServerUrl, theme.backgroundImage)
-      .then(backgroundImage => {
-        if (backgroundImage != null && this.state.backgroundImage != null) {
-          // @ts-expect-error
-          if (backgroundImage.uri === this.state.backgroundImage.uri) {
-            return
-          }
-        }
-        this.setState({ backgroundImage })
-      })
-      .catch(e => this.setState({ backgroundImage: theme.backgroundImage }))
+  })
+  React.useEffect(() => {
+    checkForUpdates().catch(error => showError(error))
+  }, [checkForUpdates])
+
+  // ---------------------------------------------------------------------
+  // Handlers
+  // ---------------------------------------------------------------------
+
+  const parentButton = React.useMemo(
+    () => ({
+      callback() {
+        Keyboard.dismiss()
+        showHelpModal()
+      },
+      text: lstrings.string_help
+    }),
+    []
+  )
+
+  const onComplete = useHandler(() => {
+    navigation.navigate('gettingStarted', {})
+  })
+
+  const onLogin = useHandler(async (account: EdgeAccount, touchIdInfo: GuiTouchIdInfo | undefined) => {
```

Comment:
Handlers should be named like `handleLogin`? I'm okay with this rename as an additional commit.

Reply:
Thanks for doing the extra cleanup.